integration-tests/api/__tests__/admin/order-edit.js

```diff
@@ -2700,6 +2700,163 @@ describe("/admin/order-edits", () => {
     })
   })
 
+  describe("Preserve custom adjustments on items change", () => {
+    let product
+    let product2
+    const orderId = IdMap.getId("order-1")
+    const prodId1 = IdMap.getId("product-1")
+    const prodId2 = IdMap.getId("product-2")
+    const lineItemId1 = IdMap.getId("line-item-1")
+
+    beforeEach(async () => {
+      await adminSeeder(dbConnection)
+
+      product = await simpleProductFactory(dbConnection, {
+        id: prodId1,
+      })
+
+      product2 = await simpleProductFactory(dbConnection, {
+        id: prodId2,
+      })
+
+      await simpleOrderFactory(dbConnection, {
+        id: orderId,
+        email: "test@testson.com",
+        region: {
+          id: "test-region",
+          name: "Test region",
+          tax_rate: 12.5,
+        },
+        discounts: [
+          {
+            code: "DUMMY_DISCOUNT_TO_TRICK_TOTALS_SERVICE",
+            rule: {
+              type: "fixed",
+              allocation: "total",
+              value: 0,
```

Comment:
self-review: it seems that custom line item adjustments won't be calculated in order totals if the order doesn't have a non free shipping discount associated.

https://github.com/medusajs/medusa/blob/8c08d003198b94c00f8428a51c0e79d2ca9d1dc7/packages/medusa/src/services/totals.ts#L997:L1005

Reply:
That is not intented, so we might as well include a fix in this one. As far as I can see, we can just remove this check entirely, and the functionality should work as expected. 

Though, that means we might do unnecessary calculations in (few) cases, but nothing to be worried about re performance.

Reply:
And by this check, I refer to these lines of code:
```
    // we only support having free shipping and one other discount, so first
    // find the discount, which is not free shipping.
    const discount = cartOrOrder.discounts?.find(
      ({ rule }) => rule.type !== DiscountRuleType.FREE_SHIPPING
    )


    if (!discount) {
      return 0
    }
```

Reply:
@olivermrbl @fPolic if we are using the orderService.decorateTotals, I believe that it should work out of the box.
The decorateTotals retrieve the context using totalsService.getCalculationContext which does what oli is [saying ](https://github.com/medusajs/medusa/blob/8c08d003198b94c00f8428a51c0e79d2ca9d1dc7/packages/medusa/src/services/totals.ts#L452) then in the method it self we are doing [that](https://github.com/medusajs/medusa/blob/8c08d003198b94c00f8428a51c0e79d2ca9d1dc7/packages/medusa/src/services/order.ts#L1785) where the discount_total as been attached [here](https://github.com/medusajs/medusa/blob/8c08d003198b94c00f8428a51c0e79d2ca9d1dc7/packages/medusa/src/services/new-totals.ts#L175). 
In the end I have tow points:
- We should fix the legacy total service as oil mentioned (for the backward compact)
- When we work on a place that rely on the legacy totals service it should be migrated to the new totals if it is straightforward as part of the good camper principle
If I am right, this is fixed by your other pr on the totals for the order edit where we agreed to migrate to a new approach. is that correct?

Reply:
> When we work on a place that relies on the legacy totals service it should be migrated to the new totals if it is straightforward as part of the good camper principle

I agree 👍 

> I believe, the get calculation context must be fixed the same way otherwise the allocation will be wrong

Just to make sure, you are saying that we should fix [this](https://github.com/medusajs/medusa/blob/8c08d003198b94c00f8428a51c0e79d2ca9d1dc7/packages/medusa/src/services/totals.ts#L455) condition the same way as well?

Reply:
Yeah since it is the same one, wdyt? I would like your opinion too :D

Reply:
Good catch - let me know wyt about this [commit](https://github.com/medusajs/medusa/pull/3085/commits/50c68095d41fa5b38cbadab3dad38c7c25346792)